python/lsst/atmospec/nightAnalysis.py

```diff
@@ -78,56 +73,61 @@ def getLineValue(table, lineName, columnName, nanForMissingValues=True):
+            self.butler = dafButler.Butler(rerunDir, instrument=instrument, collections=collections)
         self._loadExtractions(ignoreSeqNums)
         # xxx maybe just load everything and then call removeSeqNums()?
 
     def isDispersed(self, seqNum):  # TODO: change this to use the isDispersedDataId in utils
         """Match object and check is dispersed"""
-        filt = self.butler.queryMetadata('raw', 'filter', dayObs=self.dayObs, seqNum=seqNum)[0]
+        where = "exposure.day_obs = dayObs and exposure.seq_num = seqNum"
+        records = list(self.butler.registry.queryDimensionRecords("exposure",
+                                                                  where=where,
+                                                                  bind={"dayObs": self.dayObs,
+                                                                        "seqNum": seqNum},
+                                                                  instrument=self.instrument,
+                                                                  ))
+        filt = records[0].physical_filter
         grating = filt.split(FILTER_DELIMITER)[1]
         if "ronchi" in grating:
             return True
         return False
 
-    def _expIdFromSeqNum(self, seqNum):
-        return self.butler.queryMetadata('raw', 'expId', dayObs=self.dayObs, seqNum=seqNum)[0]
-
-    def _getTableFilename(self, seqNum):
-        """Return the table filename, or None if not found for the dataId"""
-        expId = self._expIdFromSeqNum(seqNum)
-        try:
-            tablePath = self.butler.getUri('spectractorOutputRoot', expId=expId)
-        except butlerExceptions.NoResults:
-            return None
-
-        tableFilename = os.path.join(tablePath, 'extractedLines.fits')
-        return tableFilename
-
     def _readOneExtractionFile(self, seqNum):
-        filename = self._getTableFilename(seqNum)  # whole thing should really just be a butler.get()
-        if not filename:
+        datasetType = "extraction"
+        try:
+            return self.butler.get(datasetType,
+                                   seq_num=seqNum,
+                                   day_obs=self.dayObs,
+                                   instrument=self.instrument)
+        except LookupError:
             return None
 
-        if os.path.exists(filename):
-            table = Table.read(filename)
-            return table
-        return None
-
     def _loadExtractions(self, ignoreSeqNums):
-        allSeqNums = self.butler.queryMetadata('raw', 'seqNum', dayObs=self.dayObs, object=self.targetName)
-
+        # Get all the observations for the night.
+        where = "exposure.day_obs = dayObs and exposure.target_name = targetName"
+        records = self.butler.registry.queryDimensionRecords("exposure",
+                                                             where=where,
+                                                             instrument=self.instrument,
```

Comment:
Can be removed once we just set to LATISS.

Reply:
It looks like setting the instrument in the butler constructor is enough and will be used automatically by the record query.